packages/remix-dev/config/flat-routes.ts

```diff
@@ -33,6 +33,10 @@ interface RouteInfo extends ConfigRoute {
   segments: string[];
 }
 
+// prettier-ignore
+let routeRegex = /(([+][/\\][^/\\:?*]+)|[/\\](index|route)|(_[^/\\:?*]+))\.(tsx?|jsx?|mdx?)$$/;
```

Comment:
i understand... some of this regex...

https://github.com/remix-run/remix/pull/5034#discussion_r1065930461